compiler/rustc_query_system/src/tls.rs

```diff
@@ -28,12 +28,6 @@ pub struct ImplicitCtxt<'a> {
     task_deps: Option<&'a Lock<TaskDeps>>,
```

Comment:
Could these `Option` be removed and instead clear the `ImplicitContext` in the tls when they would be `None`?

Reply:
I don't think so. The implicitctxt is still required to detect cycles and to print query stack even when the dependencies are ignored.

Reply:
And the other fields? At least one can be made non-optional, right? I don't see what the difference between no implicitctxt and an empty implicitctxt would be.

Reply:
Indeed, query should not need to be an option.